postprocessing/janus-pp-rec.c

```diff
@@ -1114,6 +1156,57 @@ int main(int argc, char *argv[])
 		exit(1);
 	}
 
+	/* Run restamping */
+	if(!video && !data && restamp_multiplier > 0) {
+		tmp = list;
+		uint64_t restamping_offset = 0;
+		double restamp_threshold = (double) restamp_min_th/1000;
+		double restamp_jump_multiplier = (double) restamp_multiplier/1000;
+
+		while(tmp) {
+			uint64_t original_ts = tmp->ts;
+
+			/* Update ts with current offset */
+			if(restamping_offset > 0) {
+				tmp->ts += restamping_offset;
+			}
+
+			/* Calculate diff between time of reception and the rtp timestamp */
+			double current_latency = get_latency(tmp, rate);
+
+			if(current_latency > restamp_threshold) {
+				/* Check for possible jump compared to previous latency values */
+				double moving_avg_latency = get_moving_average_of_latency(tmp->prev, rate, restamp_packets);
+				JANUS_LOG(LOG_VERB, "latency=%.2f mavg=%.2f\n", current_latency, moving_avg_latency);
+
+				/* Found a new jump in latency? */
+				if(current_latency > (moving_avg_latency*restamp_jump_multiplier)) {
+					/* Increase restamping offset with current offset */
+					restamping_offset += (current_latency-moving_avg_latency)*rate;
+
+					/* Calculate new_ts with new offset */
+					uint64_t new_ts = original_ts+restamping_offset;
+
+					/* Checking new_ts against previous ts to avoid converting with wrong timestamps */
+					if(new_ts < tmp->prev->ts) {
```

Comment:
Could you clarify a bit more this check? Are you trying to deal with overflow in the new ts?

Reply:
Actually... I am not sure we still need this one.

I put it in there while developing. It was a failsafe. I noticed that if there is a timestamp that is lower than the previous timestamp before it then the conversion will be faulty and result in huge (possibly 40GB) files.
This prevented those cases by throwing an error if I tried to create a timestamp that was lower then the previous one.

But I think that you can't get here with the current conditions so I'll remove this.